tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/BasicPublishIntegration.cs

```diff
@@ -51,23 +53,22 @@ public class BasicPublishIntegration
             where TBody : IBody, IDuckType // Versions < 6.0.0: TBody is byte[] // Versions >= 6.0.0: TBody is ReadOnlyMemory<byte>
             where TTarget : IModelBase
         {
-            var scope = RabbitMQIntegration.CreateScope(Tracer.Instance, out RabbitMQTags tags, Command, spanKind: SpanKinds.Producer, exchange: exchange, routingKey: routingKey, host: instance?.Session.Connection.Endpoint.HostName);
-            Tracer.Instance.CurrentTraceSettings.Schema.RemapPeerService(tags);
+            var scope = RabbitMQIntegration.CreateScope(Tracer.Instance, out var tags, Command, spanKind: SpanKinds.Producer, exchange: exchange, routingKey: routingKey, host: instance?.Session?.Connection?.Endpoint?.HostName);
 
-            if (scope != null)
+            // Tags is not null if span is not null, but keep analysis happy, as there's no attribute for that
+            if (scope != null && tags is not null)
             {
+                Tracer.Instance.CurrentTraceSettings.Schema.RemapPeerService(tags);
```

Comment:
same comment as above but feel free to ignore as I understand why you did it as it removes two extra if